tools/InstallerSpec/InstallerSpec/Program.cs

```diff
@@ -0,0 +1,141 @@
+            {
+                serializer.Serialize(fs, this);
+            }
+        }
+
+        public static IEnumerable<ModuleSpec> GetModules(DirectoryInfo dir)
+        {
+            foreach (var d in dir.EnumerateDirectories())
+            {
+                var modules = GetModules(d);
+                foreach (var m in modules)
+                {
+                    yield return m;
+                }
+            }
+
+            foreach (FileInfo item in dir.EnumerateFiles())
+            {
+                var info = FileVersionInfo.GetVersionInfo(item.FullName);
+                yield return new ModuleSpec() { Name = item.Name, FilePath = item.FullName.Replace(DynamoCoreVersion.BaseDirectory, @".\"), CopyForInstaller = true, DigitalSignature = Sign(info), Author = info != null ? info.CompanyName : "" };
+            }
+        }
+
+        private static bool Sign(FileVersionInfo info)
+        {
+            if(info != null && info.FileVersion != null)
+            {
+                return info.FileVersion.StartsWith(DynamoCoreVersion.BaseVersion);
+            }
+
+            return false;
+        }
+    }
+
+    class DynamoCoreVersion
+    {
+        public static string BaseVersion { get; private set; }
+        public static string BaseDirectory { get; private set; }
+        
+        public static void InitVersion(string binpath)
+        {
+            BaseDirectory = binpath;
+            if (!BaseDirectory.EndsWith(@"\"))
+                BaseDirectory += @"\";
+            string path = Path.Combine(binpath, "DynamoCore.dll");
+            if (!File.Exists(path))
+                return; //File is not available
+
+            var info = FileVersionInfo.GetVersionInfo(path);
+            BaseVersion = info.FileVersion.Substring(0, info.FileVersion.Length - 2); //drop last two characters
+        }
+    }
+
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            int nArgs = args.Length;
+            if(nArgs == 0)
+            {
+                Console.WriteLine("============== How to use this tool? ===========");
+                Console.WriteLine("Generates installer spec xml file to be used to create installer and digital signature based on a bin folder.");
+                Console.WriteLine("InstallerSpec.exe [bin folder] [xml file]");
+                return;
+            }
+            var binpath = args[0];
+            if(!Directory.Exists(binpath))
+            {
+                Console.WriteLine("===================== Error =====================");
+                Console.WriteLine("Wrong bin folder path");
+                return;
+            }
+            var filePath = @"InstallSpec.xml";
+            if(nArgs > 1)
+            {
+                filePath = args[1];
+            }
+
+            DynamoCoreVersion.InitVersion(binpath);
+            var installspec = DynamoInstallSpec.CreateFromPath(binpath);
+            installspec.Save(filePath);
```

Comment:
If I put myself in the shoes of API consumer, here's what I would expect to do:

```csharp
var version = new DynamoCoreVersion(binpath);
var installspec = DynamoInstallSpec.FromVersion(version);
installspec.Save(filePath);
```

But then that doesn't quite tell the right story (e.g. why do we create install spec given just the version?), so here's what I expect to see instead:

```csharp
// Versioning is done inside "FromPath" method call.
var installspec = DynamoInstallSpec.FromPath(binpath);
installspec.Save(filePath);
```

And then the question becomes: why do we have a static class `DynamoCoreVersion` that deals with versioning? Why isn't it part of `DynamoInstallSpec` class in anyway?